src/Pagination.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (value: number) => void;
+};
+
+const getNumberOfPages = (
+  total: number,
+  perPage: number,
+): number[] => {
+  const allPages: number[] = [];
+
+  for (let i = 1; i <= total; i += 1) {
+    allPages.push(i);
+  }
+
+  return allPages.slice(0, Math.ceil(allPages.length / perPage));
+};
+
+const getNumberOfElements = (
+  currentPage: number,
+  perPage: number,
+  total: number,
+): number[] => {
+  const elementsOnPage: number[] = [];
+
+  const startElement = (currentPage - 1) * perPage + 1;
+  const endElement = Math.min(currentPage * perPage, total);
+
+  for (let i = startElement; i <= endElement; i += 1) {
+    elementsOnPage.push(i);
+  }
+
+  return elementsOnPage;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = getNumberOfPages(total, perPage);
+  const elementsOnPage = getNumberOfElements(currentPage, perPage, total);
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === Math.max(...totalPages);
+
+  const onPageLinkClick = (page: number) => {
+    if (page !== currentPage) {
```

Comment:
It's should be okay, if you remove this `if`. Because your page will be the same and react won't rerender components